### PR TITLE
Re-pin redis version to fix freezing issue

### PR DESCRIPTION
**Pull request by dralley**
[noissue]

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
     'pygtrie~=2.3.2',
     'psycopg2>=2.7,<2.9',
     'PyYAML>=5.1.1,<5.4.0',
+    'redis~=3.1.0',
```
Comment (bmbouter) [inline setup.py]: Can we have an issue tracking the defect (even a 1-liner to have the issue number), and have a comment here linking to it?

**Comment by mdellweg** [comment]
@gmbnomis may have found a solution: https://github.com/andymccurdy/redis-py/issues/1138#issuecomment-576924979

File: setup.py (hunk 1 above)
Comment (mdellweg) [reply setup.py]: https://pulp.plan.io/issues/6038

Comment (bmbouter) [reply setup.py]: @dralley can you add the comment here also? I'm concerned that without it someone could upgrade and subtly re-introduce the issue. Wdyt?

**Comment by mdellweg** [comment]
Thank you!

**Pull request merged**